### DISCUSSION TITLE
Emails don't send on Vercel

### DISCUSSION THREAD

**Discussion by jacobsamo** [nextauthjs/next-auth#9148] [Help]
So I am using Resend as my email provider, when a user signs in with their email it will send them the email with the code to sign in or the signing link. 

these emails are sent in both dev and prod environments locally, but won't send on Vercel deployments.

these are the vercel log:
```json
  {
    "TimeUTC": "2023-11-15 09:31:24",
    "timestampInMs": 1700040684034,
    "requestPath": "www.meally.com.au/api/auth/signin/email",
    "requestMethod": "POST",
    "responseStatusCode": -10,
    "requestId": "syd1::syd1::7mqhv-1700040673960-095158d1f055",
    "requestUserAgent": [
      "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"
    ],
    "level": "error",
    "environment": "production",
    "branch": "master",
    "vercelCache": "-",
    "type": "lambda",
    "function": "/api/auth/[...nextauth]",
    "deploymentId": "dpl_Fw4xNjZnsFyQPWbP3sHwL4vrrLAK",
    "lambdaDurationInMs": -1,
    "lambdaRegion": "syd1",
    "lambdaMaxMemoryUsed": -1,
    "lambdaMemorySize": -1,
    "message": "Task timed out after 10.05 seconds\n",
    "projectId": "prj_QZSATJpQEdlMJhwMcZbI2XaNL03r",
    "projectName": ""
  },
```

This is my config: 
```ts
EmailProvider({
      server: `smtp://resend:${env.RESEND_API_KEY}@smtp.resend.com:465`,
      from: "cook@meally.com.au",
      maxAge: 24 * 60 * 60, // How long email links are valid for (default 24h)
      async generateVerificationToken() {
        const digits = "0123456789";
        let verificationCode = "";

        for (let i = 0; i < 5; i++) {
          const randomIndex = Math.floor(Math.random() * digits.length);
          verificationCode += digits.charAt(randomIndex);
        }

        return verificationCode;
      },
      async sendVerificationRequest({ identifier, url, token }) {
        console.log("Sending verification request to: ", identifier);
        await sendEmail({
          email: identifier,
          subject: "Your Meally login link",
          react: LoginLink({ url, email: identifier, token: token }),
        });

        if (process.env.NODE_ENV === "development") {
          console.log(`Login link: ${url}, token: ${token}`);
          return;
        }
      },
    }),
``` 

this is the sendEmail component
```ts
import { JSXElementConstructor, ReactElement } from "react";
import { Resend } from "resend";

import { env } from "@/env.mjs";

export const resend = env.RESEND_API_KEY
  ? new Resend(env.RESEND_API_KEY)
  : null;

export const sendEmail = async ({
  email,
  subject,
  react,
  test,
}: {
  email: string;
  subject: string;
  react: ReactElement<any, string | JSXElementConstructor<any>>;
  test?: boolean;
}) => {
  if (!resend) {
    console.error(
      "Resend is not configured. You need to add a RESEND_API_KEY in your .env file for emails to work."
    );
    return Promise.resolve();
  }
  return resend.emails.send({
    from: "cook@meally.com.au",
    to: test ? "delivered@resend.dev" : email,
    subject,
    react,
  });
};
```

**Comment by morganplouchart** [comment]
J'ai exactement la même chose.

**Comment by paulgould** [comment]
Also the same issue, everything works fine in development/localhost, but on vercel it times outs with a 504 error. I'm using the http method (not smtp) with:

 "next-auth": "5.0.0-beta.4",
 "resend": "2.1.0-canary.1",
 
 I've tested a simple email send (based on the nextjs resend docs) to ensure that resend is working otherwise, and it does.

**Reply by paulgould** [reply]
After doing some more testing, the problem seems to relate to the time it takes to generate the magic link template; I'm using react email to style the email and I found if I removed the footer component that I was importing, it would work, presumably it's saves a few seconds for whatever reason and doesn't hit the 10 second limit on the free vercel plan.

**Reply by paulgould** [reply]
I've solved the problem I was having. It turns out that the react-email library I was using made use of a Tailwind component which runs super slow on Vercel, this had the effect of pushing it over the 10 second serverless limit. I removed tailwind and now it sends in about 0.5secs as. measured by console.time. See: https://github.com/resendlabs/react-email/discussions/630

**Comment by riccardolinares** [comment]
Same issue here. It seems like sometimes it works, sometimes it doesn't... I am using an SMTP server and the custom email doesn't use tailwind css... not sure why it is happening and the debugging is very hard, since no error on the logs.

**Comment by jacobtt21** [comment]
Any answer to this??

**Comment by StruggleYang** [comment]
I had the same problem and I wanted an answer.
